src/core/queryObserver.ts

```diff
@@ -478,6 +500,23 @@ export class QueryObserver<
     // Only update if something has changed
     if (!shallowEqualObjects(result, this.currentResult)) {
       this.currentResult = result
+      const addTrackedProps = (prop: keyof QueryObserverResult) => {
+        if (!this.trackedProps.includes(prop)) {
+          this.trackedProps.push(prop)
+        }
+      }
+      this.trackedResult = {} as QueryObserverResult<TData, TError>
```

Comment:
Can we somehow only construct this object when the option is enabled? The enumerations can add up if there are a lot of observers on a page.

Reply:
yup, did that here: c653de1f394f16d0ba23bbfc66312a6b1fefb109

are there any drawbacks, like if we switch from `'tracked'` to not-tracked? I don't think so, and also doesn't sound like a use case :)